cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -387,19 +386,18 @@ data DelegationsAndRewards
 
 instance ToJSON DelegationsAndRewards where
   toJSON (DelegationsAndRewards nw delegsAndRwds) =
-      Aeson.Object $
-        Map.foldlWithKey' delegAndRwdToJson HMS.empty delegsAndRwds
+      Aeson.Array . Vector.fromList
+        . map delegAndRwdToJson $ Map.toList delegsAndRwds
     where
       delegAndRwdToJson
-        :: HashMap Text Aeson.Value
-        -> Ledger.Credential Ledger.Staking TPraosStandardCrypto
-        -> (Maybe (Hash StakePoolKey), Coin)
-        -> HashMap Text Aeson.Value
-      delegAndRwdToJson acc k (d, r) =
-        HMS.insert
-          (toKey k)
-          (Aeson.object ["delegation" .= d, "rewardAccountBalance" .= r])
-          acc
+        :: (Ledger.Credential 'Ledger.Staking TPraosStandardCrypto, (Maybe (Hash StakePoolKey), Coin))
+        -> Aeson.Value
+      delegAndRwdToJson (k, (d, r)) =
+        Aeson.object
+          [ "address" .= toKey k
```

Comment:
I would rename this `toKey` function since the resulting value is not being used as a key in a map anymore. 

Alternatively, since `toKey` isn't actually being reused, factor it out entirely and do `serialiseAddress (StakeAddress (toShelleyNetwork nw) k)`.

Reply:
Renamed to `renderAddress` .